### PR TITLE
dune integration

### PR DESCRIPTION
**Pull request by gsteenkamp89**
closes UMA-2225
## motivation

We want to fetch data from Dune to display on the site. These queries are expensive so we cache them for a day. Our Dune API key must be hidden so all this logic needs to live in a serverless function. That also means that if the cache is invalidated (after a day) and the subsequent dune call FAILS, we need to fallback to the latest result. This means we need a server side cache as well. 

## update

Moved all this logic out of an api route as it is not necessary. we are fetching this from a server component so can skip the extra network request. We still get the same caching ability. 

## TODO

- [x] make route generic (Query IDs) to we can use it for TVS on osnap page
- [ ] ~~we need server cache in case the dune API call fails. Look into using Vercel KV (basically a wrapper for Upstash/Redis)~~

UPDATE: we do not need additional caching in the case our refetch function fails, the latest stale value will be returned.

### CODE REVIEW

File: src/app/api/oev-data/route.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const API_KEY = process.env.DUNE_API_KEY!;
+const Dune = new DuneClient(API_KEY);
+
+const QUERY_ID = 3373179;
+
+export async function GET() {
+  try {
+    const executionRes = await Dune.refresh(QUERY_ID);
+    console.log(executionRes);
```
Comment (daywiss) [inline src/app/api/oev-data/route.ts:13]: stray log

File: src/lib/constants.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { DuneClient } from "@cowprotocol/ts-dune-client";
+
+// ====================================================== //
+// ====================== CONSTANTS ===================== //
+// ====================================================== //
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+export const API_KEY = process.env.DUNE_API_KEY!;
+export const Dune = new DuneClient(API_KEY);
```
Comment (daywiss) [inline src/lib/constants.ts:9]: this is a bit tricky, we should be either making API_KEY required, and asserting it, or making it optional and leaving dune: Dune | undefined. there are no hints when running the project as to whats required to resolve error if no env provided

File: src/utils/misc.ts [hunk 3]
```diff
@@ -13,3 +13,16 @@ export function addOpacityToColor(color: string, opacity: number) {
...
   return `color-mix(in srgb, transparent ${alpha}%, ${color})`;
 }
+
+export function wait(timeoutMilliseconds: number) {
+  return new Promise((res) => {
+    setTimeout(res, timeoutMilliseconds);
+  });
+}
+
+export const isDevEnvironment = process.env.NODE_ENV === "production";
```
Comment (daywiss) [inline src/utils/misc.ts:23]: what is this for

**Review by daywiss — APPROVED** [review]
looks good, just some minor things

File: src/lib/constants.ts (hunk 2 above)
Comment (gsteenkamp89) [reply src/lib/constants.ts:9]: hmm yeah excellent point!

**Pull request merged**